server/auth/auth.go

```diff
@@ -1370,6 +1389,30 @@ func (auth *AuthManager) handleMatchStatus(conn comms.Link, msg *msgjson.Message
 			return msgjson.NewError(msgjson.RPCInternalError, "DB error")
 		}
 		for _, status := range statuses {
+			var txData, txDataCoin []byte
+			var assetID uint32
+			switch {
+			case status.IsTaker && status.Status == order.MakerSwapCast: // && !status.IsMaker ?
```

Comment:
`// && !status.IsMaker ?`
Is a question? Can't be both maker and taker right?

Reply:
Should uncomment this check then to be safe.

Reply:
How to handle it correctly depends on the status of the match and how that changes the semantics of `TxData`.  This pertains to [the comment in my review](https://github.com/decred/dcrdex/pull/804#pullrequestreview-598837559):

> Regarding the TxData in MatchStatusResult, either we document very clearly that TxData can refer to different transactions depending on the match status (esp. important if user is both maker and taker) or we change to two fields, MakerTxData and TakerTxData, to avoid any ambiguity. What do you think?

The `case` here also requires `MakerSwapCast` as the match status, and if taker, that is when you want this particular tx data (the maker's raw swap tx). And whether its a self-swap or not doesn't seem to matter, but I could be mistaken.

Since we're al talking about it, I'm suspecting more strongly now that the match_status route's txdata field could use more clarity, even if that means making two separate fields and even setting them when not needed if that is clearer.

Reply:
To clarify, I think the code is right as is.  And I don't think sending lots of extra data and when it's not needed is absolutely necessary, but it could make things clearer.  Documenting the msgjson.MatchStatusResult extra clearly w.r.t. this field would also be cool.

Reply:
I'm for having 2 fields for `{M,T}akerTxData`, even though both need not be set at the same time, similar to the existing `{M,T}akerSwap`, `{M,T}akerContract` and `{M,T}akerRedeem` fields.